src/debug/nodejsDebugProvider.ts

```diff
@@ -0,0 +1,104 @@
+import * as path from "path";
+import * as vscode from "vscode";
+import { ChildProcess } from "child_process";
+
+import { IDebugProvider, PortInfo } from "./debugProvider";
+import * as config from '../components/config/config';
+import { Kubectl } from "../kubectl";
+import { IDockerfile } from "../docker/parser";
+import { kubeChannel } from "../kubeChannel";
+
+interface ExecResult {
+    readonly proxyProcess: ChildProcess;
+}
+const debuggerType = 'nodejs';
+
+export class NodejsDebugProvider implements IDebugProvider {
+    remoteRoot: String;
+    shell: string;
+
+    public getDebuggerType(): string {
+        return debuggerType;
+    }
+
+    public async isDebuggerInstalled(): Promise<boolean> {
+        return true;
+    }
+
+    public async startDebugging(workspaceFolder: string, sessionName: string, port: number): Promise<boolean> {
+
+        const debugConfiguration = {
+            type: "node",
+            request: "attach",
+            name: sessionName,
+            hostName: "localhost",
+            skipFiles: [
+                "<node_internals>/**/*.js"
+            ],
+            port
+        };
+        if (this.remoteRoot) {
+            debugConfiguration['remoteRoot'] = this.remoteRoot;
+        }
+        const currentFolder = vscode.workspace.workspaceFolders.find((folder) => folder.name === path.basename(workspaceFolder));
+        return await vscode.debug.startDebugging(currentFolder, debugConfiguration);
+    }
+
+    isSupportedImage(baseImage: string): boolean {
```

Comment:
This feels like it will cause problems when we try to find a debug provider for an image.  If this isn't the registered provider for any images then suggest returning `false` instead of throwing an exception.  cc @testforstephen